ouroboros-network/src/Ouroboros/Network/TxSubmission/Outbound.hs

```diff
@@ -69,7 +69,13 @@ data MempoolSnapshot txid tx idx =
        mempoolLookupTx   :: idx -> Maybe tx
      }
 
-data TraceTxSubmissionOutbound txid tx = TraceTxSubmissionOutbound --TODO
+data TraceTxSubmissionOutbound txid tx
+  = TraceTxSubmissionOutboundRecvMsgRequestTxs
+      ![txid]
```

Comment:
A bang on a list doesn't help that much: just the head of the list is forced to WHNF, not the tail. To be sure the list is forced, you could use `forceElemsToWHNF`, but I would not bother with that. I'd just leave out the bangs, they give a false impression of what's going on.

It would be more important if the `txids` are computed lazily, e.g., `map txId txs`, but that's not the case.

Reply:
I suppose I've gotten myself into the habit of forcing all fields to WHNF. Of course, as you mentioned, this does not guarantee that they're in NF, but I've found that it's generally not a bad idea to ensure strictness at the field-level unless laziness is specifically required. We also do the same thing for fields of `TraceEventMempool`.

However, I'm not particularly committed to this idea unless it actually helps to force to WHNF, so I'll remove the bangs and push.